log/field.go

```diff
@@ -0,0 +1,204 @@
+package log
```

Comment:
@yurishkuro while I agree we should support Reference as a first-class citizen, I was hoping to add more appropriate (for logging) Reference types before exposing that feature.